posthog/tasks/stop_surveys_reached_target.py

```diff
@@ -0,0 +1,50 @@
+from typing import List, Dict
+
+from celery import shared_task
+from django.db.models import UUIDField
+from django.utils import timezone
+
+from posthog.clickhouse.client.connection import Workload
+from posthog.client import sync_execute
+from posthog.models import Survey
+
+
+def _get_surveys_response_counts(surveys_ids: List[UUIDField]) -> Dict[str, int]:
+    data = sync_execute(
+        """
+        SELECT JSONExtractString(properties, '$survey_id') as survey_id, count()
```

Comment:
this query will time out / take too long.

There's two things you will need to do:

1. Iterate team by team (see example: https://github.com/PostHog/posthog/blob/master/posthog/models/feature_flag/flag_analytics.py#L68 )
2. Add a time range for every team query, which is the earliest start date of a running survey.

As a result of the above, we shouldn't run this task every hour, but perhaps every ~12 hours.

Reply:
actually going team by team will be overkill, and inefficient, given majority of teams don't have surveys yet, so get all surveys with limit -> group by team -> query team by team.

Reply:
Ok, makes sense, thanks for clarifying, and yeah, probably it's overkill here

> query team by team

Do you think that querying team by team is faster than querying like this:
```
        SELECT JSONExtractString(properties, '$survey_id') as survey_id, count()
        FROM events
        WHERE team_id in %(team_ids)s AND event = 'survey sent' AND survey_id in %(surveys_ids)s
        GROUP BY survey_id
```?

IMO it should be the same, but the implementation would be simpler.

Reply:
As long as it also has a time range filter, yes I think this will work.

The problem though, and I'm not sure if its worth optimising for yet, is that with multiple teams, the start date can be lower, which can significantly increase the amount of data to query, so the costs would not be the same 😅 .

Reply:
Nice, indeed!